tests/app/aws/test_s3.py

```diff
@@ -39,6 +45,27 @@ def test_get_s3_file_makes_correct_call(notify_api, mocker):
     )
 
 
+@pytest.mark.parametrize(
+    "job, job_id, job_row_number, expected_phone_number",
+    [
+        ("phone number\r\n+15555555555", "aaa", 0, "15555555555"),
+        (
+            "day of week,favorite color,phone number\r\nmonday,green,15551111111\r\ntuesday,red,15552222222",
+            "bbb",
+            1,
+            "15552222222",
+        ),
+    ],
+)
+def test_get_phone_number_from_s3(
+    mocker, job, job_id, job_row_number, expected_phone_number
+):
+    get_job_mock = mocker.patch("app.aws.s3.get_job_from_s3")
```

Comment:
When you run this test is this call actually being mocked, or is an S3 call actually happening under the hood?  This caught me by surprise with some other tests where S3 is actually hit, so I want to make sure we know for sure!

Reply:
I think it is getting mocked, because I think if I remove that line it says "failed to find job blah blah blah", and I had to fix a number of tests after my change by adding the mocking -- which is good.  I can double check.

Reply:
If it is being mocked that's awesome!  The ones I came across were not, even though it appeared like they were... 😬

Reply:
It is definitely mocked.  I always run into the mocking problem where you think you want to mock the method at the point where it is defined, but actually you want to mock it at the point where it is used.

So in test_send_to_providers, you want to mock "app.delivery.send_to_providers.get_phone_number_from_s3" which just always seems strange.

Reply:
🤔 I wonder if this is the issue with the other set of tests where it's not being mocked... either way, thank you for double checking and explaining, I appreciate it!